src/types/tests/schema-test.ts

```diff
@@ -628,6 +628,108 @@ describe('schema', () => {
     assert.deepEqual(intersection.fields, schema3.fields);
     assert.deepEqual(intersection.refinement, schema3.refinement);
   }));
+  it('tests schema union, with inline entities', async () => {
+    const manifest = await Manifest.load('./AB.schema',
+      new Loader(null, {
+        './AB.schema': `
+            schema A
+              foo: inline Foo {a: Text, b: Text}
+
+            schema B
+              foo: inline Foo {a: Text}
+            `
+      }));
+    const aType = manifest.findSchemaByName('A');
+    const bType = manifest.findSchemaByName('B');
+
+    deleteFieldRecursively(aType, 'location', {replaceWithNulls: true});
+    deleteFieldRecursively(bType, 'location', {replaceWithNulls: true});
+
+    // Union the names
+    const ABType = aType.toLiteral();
+    ABType.names = ['A', 'B'];
+
+    assert.deepEqual(Schema.union(aType, bType).toLiteral(), ABType);
+    ABType.names = ['B', 'A'];
+    assert.deepEqual(Schema.union(bType, aType).toLiteral(), ABType);
+  });
+
+  it('tests schema union, with referenced entities', async () => {
+    const manifest = await Manifest.load('./AB.schema',
+      new Loader(null, {
+        './AB.schema': `
+            schema A
+              foo: &Foo {a: Text, b: Text}
+
+            schema B
+              foo: &Foo {a: Text}
+            `
+      }));
+    const aType = manifest.findSchemaByName('A');
+    const bType = manifest.findSchemaByName('B');
+
+    deleteFieldRecursively(aType, 'location', {replaceWithNulls: true});
+    deleteFieldRecursively(bType, 'location', {replaceWithNulls: true});
+
+    // Union the names
+    const ABType = aType.toLiteral();
+    ABType.names = ['A', 'B'];
+
+    assert.deepEqual(Schema.union(aType, bType).toLiteral(), ABType);
+    ABType.names = ['B', 'A'];
+    assert.deepEqual(Schema.union(bType, aType).toLiteral(), ABType);
```

Comment:
Can we add tests for others like collections, lists, etc.?

If you refactor the testing this out you might be able to quickly add some tests. Some examples that have seen in the code base: 
- https://github.com/PolymerLabs/arcs/blob/986e1341ac7bbeda33456795931b200dd02aaade/src/tools/tests/allocator-recipe-resolver-test.ts#L766
- https://github.com/PolymerLabs/arcs/blob/986e1341ac7bbeda33456795931b200dd02aaade/src/tools/tests/allocator-recipe-resolver-test.ts#L1161

Reply:
How's that? I also filed a bug to add support for tuples of references and inline entities (which I tried to test and found that we didn't have support for)